api/src/main/java/jakarta/data/page/KeysetAwareSlice.java

```diff
@@ -145,6 +145,15 @@ public interface KeysetAwareSlice<T> extends Slice<T> {
      */
     PageRequest.Cursor getKeysetCursor(int index);
 
+    /**
+     * Returns {@code true} when it is possible to navigate to a previous
+     * page of results or if it is necessary to request a previous page in order to
+     * determine whether there are more previous results.
+     * @return {@code false} if the current page is empty or if it is known
+     *         that there is not a previous page.
+     */
+    boolean hasPrevious();
```

Comment:
Using false to both makes it unclear if it is empty or the database does not know the previous page.

Are we sure to proceed this way?

I mean, if I receive **false** I need to check anyway...

Reply:
> Using false to both makes it unclear if it is empty or the database does not know the previous page.
> 
> Are we sure to proceed this way?
> 
> I mean, if I receive **false** I need to check anyway...
> 
> Maybe an UnsupportedOperationException when it does not know

Otavio, are you sure you are reading this correctly?  When the value is `false` is when you should never try to request the page because it is known to not exist.   An empty page has no cursors and therefore cannot have a previous.  In the other case it is stated there is no previous page.  If you get `false` and then ask for `previousPageableRequest` anyways, you can expect to get an exception from the latter.  There is no point in trying.

Reply:
That was my bad. Thank you for clarifying it.